### DISCUSSION TITLE
Plans to Migrate Product.

### DISCUSSION THREAD

**Discussion by Kamsi-yonna** [coreui/coreui-free-vue-admin-template#302] [Ideas]
Hey, great work with the admin templates! It's truly a powerful tool that you've developed 👏🏿.
 I've gone through the codebase and noticed that the setup is currently in the options API, I'm wondering if there are any plans or considerations to migrate to `<script setup>` and possibly integrate TypeScript? 🤔

Migrating to `<script setup>`  and adopting TypeScript could bring significant benefits in terms of code readability 📖, maintainability 🛠, and performance 🧨. It also aligns with modern best practices and can enhance your development process, making the codebase more robust and efficient.

I would love to hear the community's ideas 💡 on this and discuss the potential migration and how it can enhance the project.

**Comment by mrholek** [comment]
@Kamsi-yonna we migrated template to `<script setup>`, regarding TypeScript. It's available in the PRO version.

**Reply by Kamsi-yonna** [reply]
Alright then
